test/test_torch.py

```diff
@@ -650,3 +653,54 @@ def test_horovod_broadcast_grad(self):
             self.assertLess(err, 0.00000001,
                             "gradient %s differs from expected %s, "
                             "error: %s" % (grad_out, expected, str(err)))
+
+    def test_load_model(self):
+        hvd.init()
+
+        N, D_in, H, D_out = 64, 100, 10, 10
+        x = torch.autograd.Variable(torch.randn(N, D_in), requires_grad=True)
+        y = torch.autograd.Variable(torch.randn(N, D_out), requires_grad=False)
+
+        def create_model():
+            model = torch.nn.Sequential(
+                torch.nn.Linear(D_in, H),
+                torch.nn.ReLU(),
+                torch.nn.Linear(H, D_out),
+            )
+
+            optimizer = torch.optim.SGD(model.parameters(),
+                                        lr=1e-4, momentum=0.9)
+
+            optimizer = hvd.DistributedOptimizer(
+                optimizer,
+                named_parameters=model.named_parameters())
+
+            return model, optimizer
+
+        model, optimizer = create_model()
+
+        y_pred = model(x)
+        loss = F.mse_loss(y_pred, y, size_average=False)
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        if hvd.rank() == 0:
+            _, fname = tempfile.mkstemp('.h5')
+            hvd.save_model(fname, model, optimizer)
```

Comment:
Can we add a custom state to the test and make sure it gets deserialized?

Reply:
Yes, will do that.